lib/firecord/repository/firebase.rb

```diff
@@ -40,21 +31,29 @@ def post(record)
       end
 
       def patch(record)
-        self.class.patch("/#{@root}/#{record.id}.json", payload(record))
+        self
+          .class.patch("/#{@root}/#{record.id}.json", payload(record))
+          .as { |response| sanitize(response) }
       end
 
       def delete(id)
-        self.class.delete("/#{@root}/#{id}.json", @options)
+        self
+          .class.delete("/#{@root}/#{id}.json", options)
+          .as { |response| response.nil? ? true : false }
       end
 
       private
 
+      def sanitize(record, default = {})
+        record.nil? ? nil : symbolize_keys(record, default)
```

Comment:
Firecord::Repository::Firebase#sanitize performs a nil-check

[Read more about it here.](https://github.com/troessner/reek/blob/master/docs/Nil-Check.md)